Creates content connectors page in security docs (#2113)
elastic/docs-content: solutions/security/get-started/content-connectors.md

---
applies_to:
  stack: preview 9.1
  serverless: preview
---

# Content connectors

> Elastic's content connectors allow you to extract, transform, index, and sync data from third-party applications including Github, PagerDuty, Jira, OpsGenie, Teams, Google Drive, Slack, email, and more. To see a complete list of the available connectors, follow the setup instructions below. 

Reviewer: ```suggestion
Elastic's content connectors allow you to extract, transform, index, and sync data from third-party applications including Github, PagerDuty, Jira, OpsGenie, Teams, Google Drive, Slack, email, and more. [View all available connectors](elasticsearch://reference/search-connectors/index.md)
```
Link to list of connectors

## Setup 

> {{stack}} supports two deployment methods: Elastic managed, and self-managed. {{sec-serverless}} only supports Elastic managed deployments. Self-managed deployments require you to manage the {{elastic-agent}} that forwards data to Elastic and allow you to customize the connector's code, whereas Elastic managed deployments use agentless technology and do not allow customization.

Reviewer: ```suggestion
{{stack}} supports two deployment methods for your connectors: Elastic managed, and self-managed. {{sec-serverless}} only supports Elastic managed connectors. Self-managed connectors require you to manage the {{elastic-agent}} that forwards data to Elastic and allow you to customize the connector's code, whereas Elastic managed connectors use agentless technology and do not allow customization.
```
Suggestion to avoid confusion with stack deployment models

Reading this paragraph, I also wonder about what this page states: https://www.elastic.co/docs/reference/search-connectors/

"As of Elastic 9.0, managed connectors on Elastic Cloud Hosted are no longer available. All connectors must be [self-managed](https://www.elastic.co/docs/reference/search-connectors/self-managed-connectors)." 

If I reconcile this paragraph with the sentence from that other page, that gives:
- Serverless only supports Elastic managed connectors
- ECH only supports self-managed connectors
- What about other Stack deployment types?

I think we should try to clarify, as I either misunderstood, or we're giving conflicting/partial information in our various pages.

@leemthompo maybe you can help here?

Reviewer 2: tldr is **self**-managed connectors can send data to Elasticsearch instances wherever they may be deployed

Reviewer 2: I don't think this paragraph belongs under **Setup** btw

> {{stack}} supports two deployment methods: Elastic managed, and self-managed. {{sec-serverless}} only supports Elastic managed deployments. Self-managed deployments require you to manage the {{elastic-agent}} that forwards data to Elastic and allow you to customize the connector's code, whereas Elastic managed deployments use agentless technology and do not allow customization.

Reviewer 2: ```suggestion
{{stack}} supports two deployment methods: Elastic managed, and self-managed. {{sec-serverless}} only supports Elastic managed deployments. Self-managed deployments require you to deploy the connector service on your own infrastructure. Self-managed connector deployments enable you to customize the connector's code, whereas Elastic managed deployments do not allow customization.
```

---
applies_to:

>   stack: preview 9.1
>   serverless: preview

Reviewer 2: ```suggestion
  stack: 
  serverless:
```

because this is now an overview page and also mentions self-managed connectors that have been around for a long time, we have to keep this page-level tag very broad and just use it to imply that there is both stack and serverless relevant info

---

# Content connectors

Elastic's content connectors allow you to extract, transform, index, and sync data from third-party applications including Github, PagerDuty, Jira, Teams, Google Drive, Slack, email, and more ([view all connectors](elasticsearch://reference/search-connectors/index.md)).

{{stack}} supports two deployment methods for your connectors: Elastic managed, and self-managed. Self-managed connectors require you to deploy on your own infrastructure (for example, run some Python code on a server you manage). When you use an Elastic managed connector, Elastic runs the infrastructure for you. Self-managed connectors can be customized, whereas Elastic managed connectors do not allow customization. 

> {{sec-serverless}} only supports Elastic managed connectors. {{stack}} deployments support either self-managed or Elastic managed deployments.

Reviewer: That's ok to leave out of scope of this issue for now, but I see here a good opportunity for a small comparison table to make it easier to understand the differences between both modes.
If you'd rather not do it right now, please create an issue :)

Reviewer 2: The info should be good per https://github.com/elastic/docs-content/pull/2113#discussion_r2244750819 at least, but yeah a table might be better format
